lib/internal/cluster/round_robin_handle.js

```diff
@@ -10,115 +10,100 @@ const net = require('net');
 const { sendHelper } = require('internal/cluster/utils');
 const { constants } = internalBinding('tcp_wrap');
 
-module.exports = RoundRobinHandle;
-
-function RoundRobinHandle(key, address, port, addressType, fd, flags) {
-  this.key = key;
-  this.all = new Map();
-  this.free = [];
-  this.handles = [];
-  this.handle = null;
-  this.server = net.createServer(assert.fail);
-
-  if (fd >= 0)
-    this.server.listen({ fd });
-  else if (port >= 0) {
-    this.server.listen({
-      port,
-      host: address,
-      // Currently, net module only supports `ipv6Only` option in `flags`.
-      ipv6Only: Boolean(flags & constants.UV_TCP_IPV6ONLY),
+class RoundRobinHandle {
+  constructor(key, address, { port, fd, flags }) {
+    this.key = key;
+    this.all = new Map();
+    this.free = new Map();
+    this.handles = [];
+    this.handle = null;
+    this.server = net.createServer(assert.fail);
+    this.attachListener(fd, port, address, flags); // UNIX socket path.
```

Comment:
Since this method is only used here, I think it's best to just keep the original code here.

Reply:
Isn't now it is giving it much more meaning to what the block is actually doing, and also if there are more case logic in some block, better to abstract it in some other place. This is done because in the future also there might be some more conditionals around that code.

Reply:
I think if the logic gets that complicated then we can revisit it at that point. Separating out 10 lines of code does not seem worthwhile.

Reply:
I still think that decreasing the vertical complexity gives the code a nice look and meaningfulness. I am still not clear about why moving it to previous implementation gives it verbosity and meaningfulness.